### DISCUSSION TITLE
Cache certain dns return code

### DISCUSSION THREAD

**Discussion by zivender** [DNSCrypt/dnscrypt-proxy#2683] [Q&A]
Hello,

As I noticed, dnscrypt-proxy will do caching if return code (status) from resolver equals to NOERROR, NXDOMAIN, or NOTAUTH. 

Is it possible to also cache SERVFAIL and REFUSED ? I am thinking of caching the response when dnssec validation is failed and when the query is blocked by a filtering resolver.

Or is it feasible to have this customized in the configuration file ?

**Comment by jedisct1** [comment]
It can be useful when only one server is used, but this is not very common.

These error codes are returned when a server is slow, overloaded or does rate limiting.

Not caching this allows another server to be used, and hopefully that one will work and it will be transparent to users. 

Even when DNSSEC validation fails with a server, it may work with another one that has working keys, or doesn't validate DNSSEC.

**Reply by zivender** [reply]
You are right on this point.

We use cloudflare dns gateway to perform filtering based on specific criterias. Unfortunately there are not many reliable options out there for this. That is the reason we can only use one server at this time.
